File: web-client/src/components/ChatPage/Message.js
import './Message.css'

const Message = ({message, user}) => {
    return (
        <div className={"message message-" + (message.sent ? "left" : "right")}>
            <div className="message-bubble">
                    <div className="message-text">
                        <p>{message.content}</p>
                    </div>
                <div className="message-timestamp">{new Date(message.created).toLocaleTimeString("en-US", {
                    hourCycle: 'h23',
                    hour: "numeric",
                    minute: "numeric"
                })}</div>
            </div>
        </div>
    );
};

export default Message;

File: class-library/Message.cs
using System.ComponentModel.DataAnnotations;

namespace class_library;

public class Message
{
    public Message(string message, string sender, DateTime date)
    {
        Text = message;
        Sender = sender;
        Timestamp = date;
    }

    public Message()
    {
    }

    [Key] public int Id { get; set; }
    [Required] public string Sender { get; set; }
    [Required] public string Text { get; set; }
    [Required] public DateTime Timestamp { get; set; }
}